### PR TITLE
Add TreePickerNode component

### PR DESCRIPTION
**Pull request by camjc**
- Modularise more variables for greater reuse.

https://trello.com/c/TYreuTXI/4730-alexandria-tree-picker-node

screen shot 2015-12-02 at 4 33 58 pm

### CODE REVIEW

File: test/components/alexandria/TreePickerNodeComponentTest.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
+/* eslint-env node, mocha */
+/* global expect */
+/* eslint no-console: 0 */
```
Comment (etaoins) [inline test/components/alexandria/TreePickerNodeComponentTest.js]: What's this for? We shouldn't use the console in tests, should we?

**Comment by etaoins** [comment]
LGTM :christmas_tree:

File: test/components/alexandria/TreePickerNodeComponentTest.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    const component = createComponent(TreePickerNodeComponent, {node: newYorkNode, includeNode});
+    const rowElement = component.props.children;
+
+    const buttonLastCellElement = rowElement.props.children[3];
+    const buttonElement = buttonLastCellElement.props.children;
+    buttonElement.props.onClick();
+    expect(nodes).to.deep.equal([
+      {
+        id: 1,
+        label: 'New York',
```
Comment (jharrington22) [inline test/components/alexandria/TreePickerNodeComponentTest.js]: Use newYorkNode

File: test/components/alexandria/TreePickerNodeComponentTest.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    const rowElement = component.props.children;
+
+    const buttonLastCellElement = rowElement.props.children[3];
+    const buttonElement = buttonLastCellElement.props.children;
+    expect(() => {
+      buttonElement.props.onClick();
+    }).to.throw('Alexandria TreePickerNode needs an includeNode handler');
+  });
+
+  it('should render a provided node', () => {
```
Comment (jharrington22) [inline test/components/alexandria/TreePickerNodeComponentTest.js]: Redundant.

File: test/components/alexandria/TreePickerNodeComponentTest.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+    const costCellElement = rowElement.props.children[2];
+    expect(costCellElement.props.className).to.equal('grid-component-cell');
+    expect(costCellElement.props.children).to.equal(300);
+  });
+
+  it('should render a provided node with an empty breadcrumb array', () => {
+    const node = {
+      id: 3,
+      label: 'Cameroon',
```
Comment (mcslash) [inline test/components/alexandria/TreePickerNodeComponentTest.js]: You spelled your name wrong ;). Also, you aren't a country. Also, I'd buy a Cameron for $400, seems like a good deal.

**Comment by mcslash** [comment]
LGTM.

File: test/components/alexandria/TreePickerNodeComponentTest.js (hunk 1 above)
Comment (camjc) [inline test/components/alexandria/TreePickerNodeComponentTest.js]: Yeah I guess not. It comes in as part of the scaffold. I guess I can remove them.

Comment (etaoins) [inline test/components/alexandria/TreePickerNodeComponentTest.js]: Stop guessing and start doing

File: src/components/alexandria/TreePickerNodeComponent.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    <span className="treepickernode-component-breadcrumbs">
+      {node.breadcrumbs.reverse().join(', ')}
+    </span> :
+    null;
+
+  const buttonElement = (
+    <span className="grid-component-cell grid-component-cell-button">
+      {
+        (_.some(selectedNodes, {id: node.id})) ?
+        <button onClick={removeNode.bind(null, node)}>Remove</button> :
```
Comment (runk) [inline src/components/alexandria/TreePickerNodeComponent.js]: I'd add indentation here

File: src/components/alexandria/TreePickerNodeComponent.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    </span>
+  );
+
+  return (
+    <span className="treepickernode-component">
+      <div className="grid-component-row">
+        {(buttonFirst) ? buttonElement : null}
+        <span className="grid-component-cell grid-component-cell-grow">
+          <span>{node.label}</span>
+          <span className="treepickernode-component-metadata"> ({node.type} in {breadcrumbsElement})</span>
```
Comment (runk) [inline src/components/alexandria/TreePickerNodeComponent.js]: should single breadcrumb element be in its own tag?

File: src/components/alexandria/TreePickerNodeComponent.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  includeNode: React.PropTypes.func.isRequired,
+  node: nodePropType.isRequired,
+  removeNode: React.PropTypes.func.isRequired,
+  selectedNodes: React.PropTypes.arrayOf(nodePropType).isRequired,
+};
+
+TreePickerNodeComponent.defaultProps = {
+  buttonFirst: false,
+  currencyFilter: (value) => value,
+  includeNode: (node) => {throw new Error(`Alexandria TreePickerNode needs an includeNode handler for ${node}`);},
```
Comment (runk) [inline src/components/alexandria/TreePickerNodeComponent.js]: Is it okay to use backquotes?

File: src/components/Main.js [hunk 8]
```diff
@@ -7,6 +7,7 @@ import Empty from 'components/alexandria/EmptyComponent';
 import React from 'react';
 import Search from 'components/alexandria/SearchComponent';
 import Slicey from 'components/alexandria/SliceyComponent';
+import TreePickerNode from 'components/alexandria/TreePickerNodeComponent';
```
Comment (runk) [inline src/components/Main.js]: '**components**/alexandria/TreePickerNode**Component**'; is a bit loud :)

File: src/components/alexandria/TreePickerNodeComponent.js (hunk 7 above)
Comment (camjc) [inline src/components/alexandria/TreePickerNodeComponent.js]: Thats how ES2015 string interpolation works. backticks and `${}`s

File: src/components/Main.js (hunk 8 above)
Comment (camjc) [inline src/components/Main.js]: Created by the scaffold and something we've done with everything else.

File: src/components/alexandria/TreePickerNodeComponent.js (hunk 6 above)
Comment (camjc) [inline src/components/alexandria/TreePickerNodeComponent.js]: nope. Turns out the name was confusing so renamed it.

**Comment by mcslash** [comment]
Green! :christmas_tree: Merging.

**Pull request merged**